### PR TITLE
Bugfix/spot many bugs

### PR DESCRIPTION
**Pull request by katherinebernstein**

### CODE REVIEW

File: src/main/kotlin/com/openlattice/authorization/listeners/PermissionMapListener.kt [hunk 1]
```diff
@@ -58,4 +58,10 @@ class PermissionMapListener(private val eventBus: EventBus) : EntryAddedListener
     override fun equals(other: Any?): Boolean {
```
Comment (makosblade) [inline src/main/kotlin/com/openlattice/authorization/listeners/PermissionMapListener.kt:58]: should regenerate this

File: src/main/kotlin/com/openlattice/conductor/rpc/ConductorConfiguration.kt [hunk 2]
```diff
@@ -21,15 +20,15 @@ data class ConductorConfiguration(
...
-        return key
-    }
 
     companion object {
         @JvmStatic
-        fun key(): ConfigurationKey {
-            return key
-        }
+        @get:JvmName("key")
+        val key = SimpleConfigurationKey("conductor.yaml")
```
Comment (makosblade) [inline src/main/kotlin/com/openlattice/conductor/rpc/ConductorConfiguration.kt:27]: can probably get rid of the key on line 11/12

File: src/main/java/com/openlattice/shuttle/EntityDefinition.java [hunk 3]
```diff
@@ -44,6 +45,7 @@
 import static com.google.common.base.Preconditions.checkNotNull;
 
 @JsonInclude( value = Include.NON_EMPTY )
+@SuppressFBWarnings( "SE_BAD_FIELD" )
```
Comment (makosblade) [inline src/main/java/com/openlattice/shuttle/EntityDefinition.java:48]: Needs reason

File: src/main/kotlin/com/openlattice/apps/services/AppService.kt [hunk 4]
```diff
@@ -118,7 +116,7 @@ class AppService(
     }
 
     fun deleteRoleFromApp(appId: UUID, roleId: UUID) {
-        val app = getApp(appId)
+        getApp(appId)
```
Comment (makosblade) [inline src/main/kotlin/com/openlattice/apps/services/AppService.kt:119]: should probably rename to something more meaningful (re: using as a null check)
Or just use
```
        requireNotNull(appId) { "App with id $appId does not exist." }
```
though actually its not nullable so. . . can just remove this line

Comment (geekbeast) [reply src/main/kotlin/com/openlattice/apps/services/AppService.kt:119]: This is just checking if app exists right? Seems like there's probably an unnecessary deep load of app going on here and we should have a method that just does containsKey on the app map.

**Review by makosblade — APPROVED** [review]

**Review by geekbeast — APPROVED** [review]

**Pull request merged**